AsyncDisplayKit/ASVideoNode.mm

```diff
@@ -437,6 +440,33 @@ - (void)setPlayerState:(ASVideoNodePlayerState)playerState
   _playerState = playerState;
 }
 
+- (void)setAssetURL:(NSURL *)assetURL
+{
+  ASDN::MutexLocker l(__instanceLock__);
+
+  if ([_asset isKindOfClass:[AVURLAsset class]] && ASObjectIsEqual(assetURL, ((AVURLAsset *)_asset).URL)) {
+      return;
+    }
+  }
+
+  [self clearFetchedData];
+
+  _asset = [AVURLAsset assetWithURL:assetURL];
+
+  [self setNeedsDataFetch];
+}
```

Comment:
It seems that `-assetURL` and `-setAssetURL:` are just convenience accessors for setting and retrieving the asset BY url. So couldn't we simplify with e.g.:

```objectivec
- (void)setAssetURL:(NSURL *)assetURL
{
   self.asset = [AVURLAsset assetWithURL:assetURL];
}
```

etc?